internal/gps/source.go

```diff
@@ -548,7 +552,7 @@ type source interface {
 	updateLocal(context.Context) error
 	listVersions(context.Context) ([]PairedVersion, error)
 	getManifestAndLock(context.Context, ProjectRoot, Revision, ProjectAnalyzer) (Manifest, Lock, error)
-	listPackages(context.Context, ProjectRoot, Revision) (pkgtree.PackageTree, error)
+	listPackages(context.Context, string, ProjectRoot, Revision) (pkgtree.PackageTree, error)
```

Comment:
yep, the cache is not designed to parameterize on a subpath. it'd be quite wasteful to inflate the cache with the additional parameter, and applying it as a post-cache transform wouldn't add much value.

better to have e.g. `func (ptree PackageTree) Subtree(subpath string) (PackageTree, error)`, which produces a new `PackageTree` containing only the subpackages along that subpath. `error` could be omitted - OTTOMH the only possible error it could express is if the provided subpath didn't match any packages in the receiver `PackageTree`.

i think this would fit in quite snugly with the other `PackageTree` methods.